src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1242,19 +1260,8 @@ impl LinkCollector<'_, '_> {
         };
 
         let resolved_self;
-        // replace `Self` with suitable item's parent name
-        let is_lone_self = path_str == "Self";
         let is_lone_crate = path_str == "crate";
-        if path_str.starts_with("Self::") || is_lone_self {
-            if let Some(ref name) = self_name {
-                if is_lone_self {
-                    path_str = name;
-                } else {
-                    resolved_self = format!("{}::{}", name, &path_str[6..]);
-                    path_str = &resolved_self;
-                }
-            }
```

Comment:
Where are paths starting with `Self::` handled in the new code? I only see explicit handling of `Self` (i.e., "lone Self"), and I'm a bit unclear about the code path `Self::` follows.

Reply:
`Self::something` paths are always resolved as type-relative associated paths (with `Self` being the type), in the same way as `TypeAlias::something`.

Reply:
Ah, I see.